c++/src/kj/async-io.c++

```diff
@@ -2764,6 +2764,121 @@ String CapabilityStreamNetworkAddress::toString() {
 
 // =======================================================================================
 
+namespace {
+
+class AggregateConnectionReceiver final: public ConnectionReceiver {
+public:
+  AggregateConnectionReceiver(Array<Own<ConnectionReceiver>> receiversParam)
+      : receivers(kj::mv(receiversParam)),
+        acceptTasks(kj::heapArray<Maybe<Promise<void>>>(receivers.size())) {}
+
+  Promise<Own<AsyncIoStream>> accept() override {
+    return acceptAuthenticated().then([](AuthenticatedStream&& authenticated) {
+      return kj::mv(authenticated.stream);
+    });
+  }
+
+  Promise<AuthenticatedStream> acceptAuthenticated() override {
+    if (backlog.empty()) {
+      auto result = kj::newAdaptedPromise<AuthenticatedStream, Waiter>(*this);
+      ensureAllAccepting();
+      return result;
+    } else {
+      auto result = kj::mv(backlog.front());
+      backlog.pop_front();
+      return result;
+    }
+  }
+
+  uint getPort() override {
+    return receivers[0]->getPort();
+  }
+  void getsockopt(int level, int option, void* value, uint* length) override {
```

Comment:
It may be used. If someone listens on `localhost:1234`, `localhost` will resolve to both `127.0.0.1` and `::1`. In the past, we'd only actually listen on the first address and ignore the second, but this PR makes us listen on both, using `AggregateConnectionReceiver`. If someone then calls `getsockopt()` on such a socket, what should happen? The implementation here maintains the old behavior.